rfcs/rdm-0064-custom-fields.md

```diff
@@ -0,0 +1,440 @@
+    "section": _("CERN Fields"),
+    "fields": [{
+        "field": "experiment",  # this should be validated against the defined fields in `RDM_RECORDS_CUSTOM_FIELDS`
+        "ui_widget": "MyCustomField",  # custom widget defined in the instance assets
+        "label": _('CERN Experiment'),
+        "placeholder": "Type an experiment...",
+        "icon": "pencil",
+        "description": _(
+            "You should fill this field with one of the experiments e.g LHC, ATLAS etc.",
+        )
+    }]
+}]
+```
+
+The instance administrator is defining a list of sections, with each section defining its own fields respectively. By using these sections, they can semantically group together different fields. The order in which sections and fields are configured is respected in the UI. All the sections and fields will be appended be default in the end of the configured form i.e upload form and community profile page.
+
+Each field can be configured with following parameters
+
+| Config property |  Type  |                          Description                          |
+| :-------------: | :----: | :-----------------------------------------------------------: |
+|      field      | string |      Should be set to one of the defined custom fields.       |
+|    ui_widget    | string |                 Name of the UI widget to use.                 |
+|      label      | string |              UI label to display for the field.               |
+|   placeholder   | string |                  Placeholder for user input.                  |
+|      icon       | string |                    Semantic UI icon name.                     |
+|   description   | string | Description to be used as helptext for the new defined field. |
+
+Moreover, any other property configured in the fields will be used as properties to initialized the respective UI widget.
+
+#### Adding or overriding UI wigets
+
+In the previous section, the `ui_widget` field was mentioned. Several of them
+are already available with InvenioRDM, however, in some cases it might be
+desired to change their behaviour (override) or use a totally new widget.
+
+The webpack loading order for the custom fields is the following:
+
+1. Try to import the widget from the RDM instance. The importer checks the cookiecutter `/assets/templates` folder. That folder is copied/symlinked when the assets are built.
+2. If the widget was not found, then it tries to import it from the local module's path.
+3. If the component is still not found, then it raises an error.
+
+With the above mechanism, instance administrators can add new files/UI widgets that can then be referenced in the config. For example, if a instance administrator wants to add a new UI widget called `MyCustomWidget` then they need to:
+
+1. Go to `/my-site/assets/templates` folder
+2. Create a new file called `MyCustomWidget.js`
+3. Use a `default` export, to export the widget
+
+```javascript
+/my-site/assets/templates/custom_fields/MyCustomWidget.js
```

Comment:
The path here does not match the explanation above, there is a `custom_field` extra folder.

Reply:
Nice catch, missing `custom_fields/` above.